### ISSUE TITLE
FNX3-16598 ⁃ Crash when wifi turns on and off

### ISSUE THREAD

**Issue by pabloag3** [mozilla-mobile/fenix#2810]
## Steps to reproduce
Turn wifi on. It eventually occur when wifi turns off too.

### Expected behavior
The app doesn't have to crash when wifi turns on or off

### Actual behavior
The app crash and need to restart the app.

![Auto Generated Inline Image 1](https://user-images.githubusercontent.com/36564545/58299702-0305bd00-7dae-11e9-840d-d4f4597b92ca.png)

* Android device: 5.0
* Fenix version: 1.0.1921

**Comment by kbrosnan** [comment]
Would you be able to get a logcat for this issue? 

* Download ADB for your computer https://developer.android.com/studio/releases/platform-tools
* Open the folder that you extracted the adb program to on the command line/terminal
* run `adb logcat -v time > crash.txt` 
* reproduce the crash
* attach the file `crash.txt` to this issue

**Comment by pabloag3** [comment]
Sure, here it is.

[crash.txt](https://github.com/mozilla-mobile/fenix/files/3216930/crash.txt)

**Comment by ekager** [comment]
I think it's this:

> 05-24 08:34:55.117 E/AndroidRuntime(17130): FATAL EXCEPTION: main
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): Process: org.mozilla.fenix, PID: 17130
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): java.lang.RuntimeException: Error receiving broadcast Intent { act=android.net.conn.CONNECTIVITY_CHANGE flg=0x4000010 (has extras) } in androidx.work.impl.constraints.trackers.NetworkStateTracker$NetworkStateBroadcastReceiver@d70de9
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:933)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at android.os.Handler.handleCallback(Handler.java:739)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at android.os.Handler.dispatchMessage(Handler.java:95)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at android.os.Looper.loop(Looper.java:145)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at android.app.ActivityThread.main(ActivityThread.java:5944)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at java.lang.reflect.Method.invoke(Native Method)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at java.lang.reflect.Method.invoke(Method.java:372)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): Caused by: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@233960b2 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@36909803[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2011)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:793)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:298)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:503)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at java.util.concurrent.Executors$DelegatedScheduledExecutorService.schedule(Executors.java:644)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at androidx.work.impl.background.systemalarm.WorkTimer.startTimer(WorkTimer.java:7)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at androidx.work.impl.background.systemalarm.DelayMetCommandHandler.onAllConstraintsMet(DelayMetCommandHandler.java:10)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at androidx.work.impl.constraints.WorkConstraintsTracker.onConstraintMet(WorkConstraintsTracker.java:8)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at androidx.work.impl.constraints.controllers.ConstraintController.updateCallback(ConstraintController.java:3)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at androidx.work.impl.constraints.trackers.ConstraintTracker.setState(ConstraintTracker.java:9)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at androidx.work.impl.constraints.trackers.NetworkStateTracker$NetworkStateBroadcastReceiver.onReceive(NetworkStateTracker.java:4)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:923)
> 
> 05-24 08:34:55.117 E/AndroidRuntime(17130): 	... 8 more

**Comment by ekager** [comment]
On the Fenix side we're not listening for any broadcasts, so I opened an AC issue for this https://github.com/mozilla-mobile/android-components/issues/3110

**Comment by kbrosnan** [comment]
Sentry issue: [FENIX-15W](https://sentry.prod.mozaws.net/operations/fenix/issues/5769134/)

**Comment by kbrosnan** [comment]
Very similar crash stack: [FENIX-VP](https://sentry.prod.mozaws.net/operations/fenix/issues/5674157/)

**Comment by AaronMT** [comment]
All you have to do is launch Fenix in Airplane mode.

```java
2019-05-27 12:10:08.653 10104-10104/org.mozilla.fenix.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix.debug, PID: 10104
    mozilla.appservices.fxaclient.FxaException$Network: Network error: Network error: java.io.IOException: "org.mozilla.geckoview.WebRequestError: Request failed, error=0x25, category=0x5"
        at mozilla.appservices.fxaclient.rust.RustError.intoException(RustError.kt:43)
        at mozilla.appservices.fxaclient.FirefoxAccount.ensureCapabilities(FirefoxAccount.kt:717)
        at mozilla.components.service.fxa.FxaDeviceConstellation$ensureCapabilitiesAsync$1.invokeSuspend(FxaDeviceConstellation.kt:88)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:238)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742)

**Comment by AaronMT** [comment]
Maybe we want to split this? Filed mine as https://github.com/mozilla-mobile/fenix/issues/2888

**Comment by sblatz** [comment]
@ekager so is this blocked on https://github.com/mozilla-mobile/android-components/issues/3110?

**Comment by softvision-miralobontiu** [comment]
Verified as fixed, on the latest build 1.0.1922 (#11530616), with Google Pixel (Android 9), and Motorola Moto G6 (Android 8).
